cli/src/config.rs

```diff
@@ -0,0 +1,131 @@
+use color_eyre::Result;
+use serde::{de::DeserializeOwned, Deserialize};
+use serde_with::{serde_as, DisplayFromStr};
+use std::{fs::File, io::BufReader, net::SocketAddr, path::PathBuf};
+use tower_http::cors::CorsLayer;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Timeout(u64);
+#[derive(Debug, Clone, Deserialize)]
+pub struct CorsAllowAll(bool);
+#[derive(Debug, Clone, Deserialize)]
+pub struct LogAmount(usize);
+#[derive(Debug, Clone, Deserialize)]
+pub struct LogSize(u64);
+
+/// Server Configuration passed on initialization
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Config {
+    /// Server address <host:port>
+    pub address: SocketAddr,
+
+    /// Enable Default Cors configuration
+    #[serde(default)]
+    pub cors: CorsAllowAll,
+
+    /// Max Time to Responce, in milliseconds
+    #[serde(default)]
+    pub request_timeout: Timeout,
```

Comment:
```suggestion
    pub request_timeout: Duration,
```

Also add `humantime_serde` to parse this duration from human readable form